tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,185 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
+				log.FailOnError(err, "error getting pool drives from node [%s]", selectedNode.Name)
+				drvs := drvMap[fmt.Sprintf("%d", selectedPool.ID)]
+				maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+				fmt.Printf("maxDrivesAllowed %v", maxDrivesAllowed)
+			} else {
+				log.FailOnError(err, "error getting pool from node [%s]", selectedNode.Name)
+			}
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				fmt.Printf("i %v", i)
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "fialed while getting pool size")
+				driveSize := drvSize * uint64(i)
+				expectedSize := (selectedPool.TotalSize / units.GiB) + driveSize
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				log.FailOnError(err, "error while expanding pool")
+				isjournal, err := isJournalEnabled()
+				log.FailOnError(err, "Failed to check is journal enabled")
+				resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+				err = Inst().V.RefreshDriverEndpoints()
+				log.FailOnError(err, "error refreshing volume endpoints")
+				dash.VerifyFatal(resizeErr != nil, false, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+				if i == maxDrivesAllowed+1 {
+					dash.VerifyFatal(resizeErr != nil, true, fmt.Sprintf("Expected new size to be '%d' or '%d'", expectedSize, expectedSize-3))
+					if selectedPool.LastOperation.Msg != "" {
+						dash.VerifyFatal(strings.Contains(selectedPool.LastOperation.Msg, "unable to add 1 new drive(s) as it would exceed maximum supported drives (6)"), true, "Error expected as drive added more than allowed per pool")
+					} else {
+						err := errors.New("failed while getting msg")
+						log.FailOnError(err, "failed while getting msg")
+					}
+				}
+			}
+		})
+
+		Step("Test max pool count", func() {
+			log.InfoD(stepLog)
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			log.FailOnError(err, "error adding new pools")
+			err = addNewPools(selectedNode, 1)
+			dash.VerifyFatal(err != nil, true, "Error expected as pools trying to create is more than allowed")
+			poolList, err = GetPoolsDetailsOnNode(selectedNode)
+		})
+
+		Step("Test max drives per node", func() {
+			log.InfoD(stepLog)
+			var selectedPools string
+			poolList, err := GetPoolsDetailsOnNode(selectedNode)
+			log.FailOnError(err, "fialed while getting updated node pool list")
+			//first add all the new pools
+			poolLeft := maxPoolLength - len(poolList)
+			err = addNewPools(selectedNode, poolLeft)
+			drvNum, err := Inst().N.RunCommand(selectedNode, "lsblk -l -d -e 11 -n -o NAME|wc -l", node.ConnectionOpts{
+				Timeout:         defaultCommandTimeout,
+				TimeBeforeRetry: defaultCommandRetry,
+				IgnoreError:     false,
+				Sudo:            false,
+			})
+			drvNum1 := strings.TrimSpace(string(drvNum))
```

Comment:
You can combine, `9866`, `9877`, `9868` into one line.

Reply:
Not understanding this comment pls let me know more